samples/app-ReporterPlus/ReporterPlus/Controllers/HomeController.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using ReporterPlus.Helpers;
+using Microsoft.Extensions.Configuration;
+using System.Collections.Generic;
+
+namespace ReporterPlus.Controllers
+{
+    public class HomeController : Controller
+    {
+        public readonly string blobAccountName;
+        public readonly string blobSasString;
+
+        public HomeController(IConfiguration configuration) : base()
+        {
+            //Initializing Constants
+            Constants.BlobContainerName = configuration["BlobContainerName"];
+            Constants.BlobConnectionString = configuration["BlobConnectionString"];
+            Constants.ServiceUrl = configuration["ServiceUrl"];
+            Constants.MicrosoftAppId = configuration["MicrosoftAppId"];
```

Comment:
We are also populating these in the `ReporterPlusBot.cs`. Why are we re-initiating the same constants here?

Reply:
This is because, when the app is run and before invoking the bot file, if the user clicks on the ViewDetails, then the bot constructor is not loaded and hence the constants value would be null, so for that sake we are just declaring these value in Controller also.